internal/auth/options.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package auth
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/pagination"
+)
+
+// The option type is how options are passed as arugments
+type Option func(*Options) error
+
+// Options is a struct that contains all the possible auth options
+type Options struct {
+	WithLimit               int
+	WithStartPageAfterItem  pagination.Item
+	WithUnauthenticatedUser bool
+}
+
+// getDefaultOptions returns a new Options struct
+func getDefaultOptions() *Options {
+	return &Options{}
+}
+
+// GetOpts loops over the options and provides
+// the configured options struct
+func GetOpts(opts ...Option) (*Options, error) {
+	newOpts := getDefaultOptions()
+	for _, opt := range opts {
+		if err := opt(newOpts); err != nil {
+			return nil, err
+		}
+	}
+	return newOpts, nil
+}
+
+// WithLimit provides an option to provide a limit.
+// If WithLimit <= 0, then default limits are used for results.
+func WithLimit(_ context.Context, l int) Option {
+	return func(o *Options) error {
+		o.WithLimit = l
+		return nil
```

Comment:
Could we error if `l < 0`?

Reply:
Hmm, we're a bit inconsistent on where we check for the limit and how we handle it across all of pagination - in some domains like auth methods, we'll error in the repository, but in other domains we just override the limit with the repository's default. I think, at the very least, we may as well keep all the checks in the repository methods to keep it consistent in location, for now. Thoughts?